lab_3/main.py

```diff
@@ -0,0 +1,307 @@
+        self.logger.debug(f'Changed sorted data: {self.sorted_data}')
+        self.save_data_to_csv(self.sorted_data, self.destination)
+
+    def save_data_to_csv(self, data, destination):
+        """
+         This method saves the sorted data to a CSV file at the destination.
+        :param data: a list of dictionaries containing data to be saved.
+        :param destination: the file path where the CSV file will be created.
+        """
+        with open(destination, 'w', newline='', encoding='utf-8') as csvfile:
+            writer = csv.DictWriter(csvfile, fieldnames=self.fieldnames)
+            writer.writeheader()
+            writer.writerows(data)
+
+        self.logger.info(f'The data was successfully saved to {destination}.')
+
+    def create_new_working_dir(self):
+        """ This method creates a new working directory and moves the sorted data file to that directory. """
+        os.mkdir(os.path.join(self.args.destination, 'Data'))
+        self.logger.info(f'The new folder was successfully created. His full path {self.new_path}')
+
+        shutil.move(self.destination, self.new_path)
+        self.logger.info(f'File with sorted data moved to {self.new_path}')
+
+        os.chdir(self.new_path)
+        self.logger.info(f'New working directory: {self.new_path}')
+
+    def rearrange_the_data(self):
+        """ This method rearranges the sorted data into a new format, grouped by decade and country. """
+        for i in self.sorted_data:
+            dob_date = datetime.strptime(i['dob.date'], '%m/%d/%Y')
+            decade = f'{dob_date.year // 10 * 10}-th'
+            country = i['location.country']
+            self.rearranged_data.setdefault(decade, {}).setdefault(country, [])
+            self.rearranged_data[decade][country].append(i)
+
+        for decade, decade_data in self.rearranged_data.items():
+            for country, data in decade_data.items():
+                self.rearranged_data[decade][country] = sorted(data, key=lambda x: (x['dob.date'],
+                                                                                    x['location.country']))
+
+        self.logger.info('The sorted date was successfully rebuilt to the new format.')
+        self.logger.debug(f'Rearranged data: {self.rearranged_data}')
+
+    def count_age(self, users_data):
+        """
+        Calculate the maximum and average age from a list of user data.
+        :param users_data: a list of dictionaries containing user data,
+                            with each dictionary having 'dob.age' as the key for age.
+        :return max_age, avg_registered: returns the calculated maximum age and average age as integers.
+        """
+        # Count max age and average age
+        ages = [int(i['dob.age']) for i in users_data]
+        max_age = max(ages)
+        total_ages = sum(ages)
+        avg_registered = int(total_ages / len(users_data))
+        return max_age, avg_registered
```

Comment:
as another option:
ages = [int(i['dob.age']) for i in users_data]
return max(ages), int(sum(ages) / len(users_data))